### PR TITLE
fix(Participant): migrate to NcListItem

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Ref #12800 
* Requires https://github.com/nextcloud-libraries/nextcloud-vue/pull/5870

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
Screenshot before | Screenshot after
image | image
image | image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Comment by nickvergessen** [comment]
That's quite "in the face" for a "list of selectable items"...

I had already mentioned back then when the components where created that it might not be the best idea to combine the items in the right sidebar with the attendee picking in the convo creation dialog.
I'd still be fine with splitting that. They operate on total different data set, have different data and are inside different context.

**Comment by Antreesy** [comment]
> I'd still be fine with splitting that

Could be a good first step, before upstream PR is here. Search result is simple enough to migrate it right now

File: src/components/RightSidebar/Participants/Participant.vue [hunk 1]
```diff
@@ -4,154 +4,128 @@
 -->
 
 <template>
-	<component :is="tag"
+	<NcListItem :name="computedName"
```
Comment (DorraJaouad) [inline src/components/RightSidebar/Participants/Participant.vue:7]:
```suggestion
	<NcListItem 
```

as you are using name slot

File: src/components/RightSidebar/Participants/Participant.vue [hunk 2]
```diff
@@ -1176,110 +1124,84 @@ export default {
...
+	}
+
+	&--offline &__user-name {
+		color: var(--color-text-maxcontrast);
+	}
 
-	&.isSearched {
+	&--searched :deep(.list-item) {
 		cursor: pointer;
 	}
```
Comment (DorraJaouad) [inline src/components/RightSidebar/Participants/Participant.vue:1146]: It is not applied and participant is now selectable in list, is it expected ?
participants

File: src/components/RightSidebar/Participants/Participant.vue [hunk 3]
```diff
@@ -1176,110 +1124,84 @@ export default {
...
-			padding-right: 5px;
-			/** call icon on the right most column */
-			width: calc(100% - 90px);
-		}
+			&:hover,
+			&:focus-within,
+			&:has(:focus-visible),
+			&:has(:active) {
+				background-color: var(--color-primary-element-light-hover);
+			}
```
Comment (DorraJaouad) [inline src/components/RightSidebar/Participants/Participant.vue:1158]: As shown above, I am not sure if it is expected to have background-color on click focus, it seems like you can do some action on it from a header menu

**Review by DorraJaouad — COMMENTED** [review]
It is almost ready 🌟

File: src/components/RightSidebar/Participants/Participant.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/RightSidebar/Participants/Participant.vue:7]: Prop is required, so test will scream about it

File: src/components/RightSidebar/Participants/Participant.vue (hunk 2 above)
Comment (Antreesy) [reply src/components/RightSidebar/Participants/Participant.vue:1146]: styles fixed.
It's expected as it is interactive element `<a/>`. we have it before anyway with `tabindex="0"`, just without visual feedback

File: src/components/RightSidebar/Participants/Participant.vue (hunk 3 above)
Comment (ShGKme) [reply src/components/RightSidebar/Participants/Participant.vue:1158]: Better to be fixed in upstream. CUrrently `NcListItem` always supposes item to be a link.

**Comment by ShGKme** [comment]
Follow-ups:
- With additional padding it is not inline with search
  image
- Structure is not a valid HTML
  image
- Placeholder is different from list item (an old issue? cc @ShGKme)
  image
- Old a11y issue - on creation Participant represent a checkbox but it is a link
  image

**Pull request merged**